bin/bootstrax

```diff
@@ -474,31 +435,45 @@ def override_target(rd):
         return False
 
 
-def set_state(state):
+def set_state(state, update_fields=None):
     """Inform the bootstrax collection we're in a different state
 
     if state is None, leave state unchanged, just update heartbeat time
     """
     ping_dbs()
 
-    if state_doc_id is None:
-        log.debug("Not updating bootstrax doc: none selected. "
-                  "You are probably running a rogue bootstrax operation...")
-        return
-    update = {'time': now()}
-    if state is not None:
-        update['state'] = state
-    bs_coll.update_one(
-        {'_id': state_doc_id},
-        {'$set': update})
-
+    if not state:
+        # Find the last message of this host
+        previous_entry = bs_coll.find_one({'host': hostname,
+                                           'time': {'$lt': now(-timeouts['bootstrax_presumed_dead'])}},
+                                          sort=[('time', pymongo.DESCENDING)])
+        if not previous_entry:
+            raise ValueError('No previous entries nor state set. Are you running a rogue bootstrax?')
```

Comment:
If the main loop responds slowly for some reason, is there a possibility that we reach this condition and crash?

Reply:
This is indeed not the best place to raise ValueErrors for something we foresee happening. I'll change it.